### PR TITLE
refactor(list): rewrite with hook

### PR DESCRIPTION
**Pull request by hengkx**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  refactor with hook         |
| 🇨🇳 Chinese |  用 react hook 重构         |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by xrkffgg** [comment]
@hengkx Rate 你做了吗？

**Comment by hengkx** [comment]
> @hengkx Rate 你做了吗？

没有你做

**Comment by xrkffgg** [comment]
ok





------------------ Original ------------------

File: components/list/style/index.tsx [hunk 1]
```diff
@@ -2,7 +2,6 @@ import '../../style/index.less';
 import './index.less';
 
 // style dependencies
-import '../../empty/style';
```
Comment (afc163) [inline components/list/style/index.tsx:5]: 这个删了单独引入 List 时是否会有问题？

Comment (hengkx) [reply components/list/style/index.tsx:5]: https://codesandbox.io/s/antd-reproduction-template-sjzsb?file=/index.js
这样看没问题
lint检测到没有使用到它

Comment (afc163) [reply components/list/style/index.tsx:5]: @zombieJ check it

Comment (zombieJ) [reply components/list/style/index.tsx:5]: 用了 Empty 就需要依赖它。depslint 用的正则匹配，需要再加个规则把它拿回来：

https://github.com/ant-design/ant-design/blob/41da9de946ede284407ffe87babfce4d8900f309/.depslintrc.js#L12

Comment (hengkx) [reply components/list/style/index.tsx:5]: 里面 有用 ，那我吧他还原 check 过不去 你们处理下吧

**Pull request merged**